src/track/beatmap.cpp

```diff
@@ -522,17 +522,17 @@ bool BeatMap::hasBeatInRange(double startSample, double stopSample) const {
     return false;
 }
 
-double BeatMap::getBpm() const {
+mixxx::Bpm BeatMap::getBpm() const {
     if (!isValid()) {
-        return mixxx::Bpm::kValueUndefined;
+        return mixxx::Bpm();
     }
     return m_nominalBpm;
 }
 
 // Note: Also called from the engine thread
-double BeatMap::getBpmAroundPosition(double curSample, int n) const {
+mixxx::Bpm BeatMap::getBpmAroundPosition(double curSample, int n) const {
     if (!isValid()) {
-        return -1;
+        return mixxx::Bpm();
```

Comment:
Either return `mixxx::Bpm()` or `{}` but not a mixture of both in various places. I would prefer the latter, because we don't need to repeat the return type multiple times.

Reply:
https://github.com/mixxxdj/mixxx/pull/4045/commits/603d3253ceba37c894fcffc304ef934a667e0437